js/adapt-pageIncompletePrompt.js

```diff
@@ -79,27 +78,31 @@ define([
         onLeaveCancel: function() {
             if (!this.inPopup) return;
             this.inPopup = false;
-            
+
             this.stopListening(Adapt, "notify:cancelled");
             this.enableRouterNavigation(true);
             this.handleRoute = true;
         },
 
         onRouterNavigate: function(routeArguments) {
-            
+
             if(!this.isEnabled() || this.allComponentsComplete()) return;
 
             this.href = window.location.href;
 
             if (routeArguments[0]) {
-                //check if routing to current page child
-                //exit if on same page
+                var id = routeArguments[0];
+                // exit if on same page (e.g. if doing 'retry assessment')
+                if (id === Adapt.location._currentId) return;
+                // check if routing to current page child
                 try {
-                    var id = routeArguments[0];
                     var model = Adapt.findById(id);
                     var parent = model.findAncestor("contentObjects");
                     if (parent.get("_id") == this.pageModel.get("_id")) return;
-                } catch (e) {}
+                } catch (e) {
```

Comment:
Is the `try...catch` still required?

Reply:
I'm not sure TBH, I'll defer to @oliverfoster on that one...

Reply:
it does catch other errors that haven't been specifically 'thrown' though - for example the issue I'm fixing with this PR was previously being caught and handled by the `catch` block - which is why I added the `return` to it as I don't think the prompt should be shown if an error was encountered at this point.